User auth implementation

dqu123 opened the pull request:
<h3>Specification:</h3>

- [x] Members can register accounts
  - [ ] Test script checks this
- [x] Login / logout (sessions and being able to have features only available to users)
  - [ ] Test script
- [ ] Account settings can be changed 
  - [ ] Test script
- [x] Password reset
  - [ ] Test script (Maybe just unit test, might be hard to test emailing automatically)

dqu123 committed:
Fixed space.

Commit:
Added link to userreg.html in navbar.

Commit:
Updated the user registration message to clarify Donut account eligibility.

Commit:
Added contact.html page.

dqu123 committed:
Merge remote-tracking branch 'origin' into user_auth

dqu123 committed:
Merge branch 'master' of https://github.com/ASCIT/donut into user_auth

dqu123 committed:
Merge branch 'master' into user_auth

dqu123 committed:
Set 2 space indent.

dqu123 committed:
Add enum34 to requirements.txt

dqu123 committed:
Add supporting hashing / reset key code from RuddockWebsite.

dqu123 committed:
Add password reset form.

dqu123 committed:
Add account module with create_account and request_account pages.

allisonkong commented:
Quick heads up - I was working on a python 3 friendly rewrite of the password hashing stuff here:
https://github.com/RuddockHouse/RuddockWebsite/blob/py3/ruddock/security/passwords.py

dqu123 committed:
Uncomment try catch block in account/helpers.py (done during debugging).

dqu123 committed:
Make get_permissions() always return an empty list to test without permissions backend.

We will have to design a different permissions system than RuddockWebsite because we need
to control voting and mailing lists on a case by case basis.

dqu123 committed:
Add forgot_password and reset_password templates.

dqu123 committed:
Change flask.flash message contact info to devteam@donut.caltech.edu.

dqu123 committed:
Fix queries to match differences in schemas (no members_extra table, name column).

dqu123 committed:
Only show 'New Account' button when not logged in.

dqu123 committed:
Add newlines to account management email templates.

dqu123 committed:
Remove unused user registration page.

dqu123 committed:
Change 'name' column to 'first_name' to reflect schema differences between Donut and Ruddock.
